objectherkenning_openbare_ruimte/convert_dataset_pipeline/source/equirectangular_to_cubemap_converter.py

```diff
@@ -0,0 +1,752 @@
+        """
+        Convert absolute corner coordinates to YOLO annotation with normalized values.
+
+        Parameters
+        ----------
+        yolo_class: str
+            The class of the object.
+        top_left : float
+            Top-left absolute corner coordinates.
+        bottom_right : float
+            Bottom-right absolute corner coordinates.
+        face_w : int
+            Width of the face.
+        face_h : int
+            Height of the face.
+
+        Returns
+        tuple
+            A tuple containing the YOLO annotation in the format:
+            "<yolo_class> <x_center_norm> <y_center_norm> <w_norm> <h_norm>".
+        -------
+
+        """
+
+        x_min, y_min = top_left
+        x_max, y_max = bottom_right
+
+        x_center_abs = (x_min + x_max) / 2
+        y_center_abs = (y_min + y_max) / 2
+        width_abs = x_max - x_min
+        height_abs = y_max - y_min
+
+        x_center_norm = x_center_abs / face_w
+        y_center_norm = y_center_abs / face_h
+        width_norm = width_abs / face_w
+        height_norm = height_abs / face_h
+
+        return yolo_class, x_center_norm, y_center_norm, width_norm, height_norm
+
+    @staticmethod
+    def _write_annotation_to_file(
+        output_path: str,
+        img_name: str,
+        face_idx: int,
+        annotation: Tuple[str, float, float, float, float],
+    ) -> None:
+        """
+        Write the converted YOLO annotation to the specified annotation file after clearing any existing content.
+
+        Parameters
+        ----------
+        output_path : str
+            The base directory where the annotation files are stored.
+        img_name : str
+            The name of the image file, used to derive the folder name for storing annotations.
+        face_idx : str
+            The index of the face, used to derive the annotation file name.
+        annotation : str
+            The converted YOLO annotation to write to the file.
+
+        Returns
+        -------
+        None
+        """
+        face_name = EquirectangularToCubemapConverter._convert_face_idx_to_name(
+            face_idx
+        )
+
+        annotation_file = os.path.join(output_path, img_name, f"{face_name}.txt")
+        converted_yolo_annotation_str = " ".join(map(str, annotation))
+
+        os.makedirs(os.path.dirname(annotation_file), exist_ok=True)
+
+        with open(annotation_file, "a") as file:
+            file.write(converted_yolo_annotation_str + "\n")
+
+        logging.info(f"Annotation written to {annotation_file}")
+
+    @staticmethod
+    def _adjust_bounding_box_corners(
+        processed_corners: Dict[str, Any], face_width: int
+    ) -> Tuple[Tuple[float, float], Tuple[float, float]]:
+        """
+        Adjust the bounding box corners coordinates based on already processed corners within a face.
+
+        This function finalizes the coordinates of a bounding box by considering the adjustments
```

Comment:
This explanation could be more clear. What does it mean for bb corners to have been "processed"? What adjustments are being done?